spec/consul_application_settings/consul_provider_spec.rb

```diff
@@ -0,0 +1,106 @@
+RSpec.describe ConsulApplicationSettings::ConsulProvider do
+  let(:provider) { described_class.new(base_path, config) }
+  let(:base_path) { '' }
+  let(:config) do
+    ConsulApplicationSettings::Configuration.new.tap do |config|
+      config.disable_consul_connection_errors = disable_consul_connection_errors
+    end
+  end
+  let(:disable_consul_connection_errors) { true }
```

Comment:
You need this functionality only for a single context. I would rather override `let(: config)` inside of that context instead of exposing it here.

Reply:
Well, then I'll have to duplicate 5 lines of code instead of 1. I would leave it if it's fine with you.

Reply:
Not five, In the root - config will be initialized in one line. 

My approach is we should not sacrifice descriptiveness in favor of DRY principles in tests. Just discovered there is a special term for this [DAMP not Dry](https://stackoverflow.com/questions/6453235/what-does-damp-not-dry-mean-when-talking-about-unit-tests). There is lot of interesting ideas if you search for DAMP, Wet tests, etc.

I will leave it up to you to make this decision

Reply:
I like "DAMP not Dry" very much. You're right about one line.

Reply:
Fixed.